src/Microsoft.DotNet.CodeFormatting/FormattingEngineImplementation.cs

```diff
@@ -33,8 +33,9 @@ internal sealed partial class FormattingEngineImplementation : IFormattingEngine
         private readonly IEnumerable<ExportFactory<ILocalSemanticFormattingRule, LocalSemanticRule>> _localSemanticRules;
         private readonly IEnumerable<ExportFactory<IGlobalSemanticFormattingRule, GlobalSemanticRule>> _globalSemanticRules;
         private readonly Stopwatch _watch = new Stopwatch();
-        private readonly Dictionary<string, bool> _ruleMap = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
-        private readonly ImmutableDictionary<string, CodeFixProvider> _fixerMap;
+        private readonly Dictionary<string, bool> _ruleEnabledMap = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
+        private readonly Dictionary<string, bool> _diagnosticEnabledMap;
+        private readonly ImmutableDictionary<string, CodeFixProvider> _diagnosticIdToFixerMap;
```

Comment:
I did some renaming here. In Sri's code he had a "fixerMap" but it served a different purpose than the rule map. I renamed all the maps to clarify their purpose.